haas/rest.py

```diff
@@ -186,18 +195,35 @@ def _do_validation(schema, kwargs):
     `ValidationError`.
     """
 
-    if flask.request.data == '':
-        # No request body
-        final_kwargs = {}
+    final_kwargs = {}
+
+    if flask.request.method == "GET":
+        # GET requests can use URL and query parameter arguments
+        if flask.request.data != '':
+            raise ValidationError("GET request made with a non-empty request"
+                                  " body")
+
+        for key,value in flask.request.args.iteritems():
+            if final_kwargs.has_key(key):
+                raise ValidationError("Parameter specified more than once")
+            elif value is None or value == '':
+                # TODO: if we want to take flags (ie - an option that has no
+                # value), change this check and add logic in the next else to
+                # take care of that. We'll also need to pick a standard value
+                # to represent a flag (like 'True', 'None' or '').
+                raise ValidationError("Empty parameter specified")
```

Comment:
I just don't see why this is a special case. The empty string is still a string, and anyway all our code is already set up to deal with input of different types. If the call expects query parameters for the GET, it *needs* to include those in it's schema (that's the whole point of this patch!) So this feels both redundant and completely arbitrary.